gs_quant/timeseries/arima.py

```diff
@@ -0,0 +1,248 @@
+from typing import Iterable, Optional, Union, Tuple
+
+import itertools
+import datetime as dt
+import pandas as pd
+import numpy as np
+from tqdm import tqdm
+from statsmodels.tsa.arima_model import ARIMA
+from statsmodels.tools.eval_measures import mse
+
+
+@dataclass
+class ARIMA_BestParams:
+    freq: str = ''
+    p: int = None
+    d: int = None
+    q: int = None
+    const: float = None
+    ar_coef: list = None
+    ma_coef: list = None
+    resid: list = None
+    series: pd.Series = None
+
+class arima():
+    """
+    ARIMA is the Autoregressive Integrated Moving Average Model and is used 
+    to normalize and forecast time series data. ARIMA has 3 parameters: (p, d, q)
+    where:
+        :p is the number of autoregressive terms
+        :d is the number of nonseasonal differences
+        :q is the number of lagged forecast errors in the prediction equation
+    
+    An ARIMA model is selected from the Catesian product of sets p, q, and d. The 
+    time series is split into train and test sets and an ARIMA model is fit for 
+    every combination on the training set. The model with the lowest mean-squared 
+    error (MSE) on the test set is selected as the best model. The original 
+    times series can then be transformed by the best model.
+
+    Autoregressive components are past values of the variable of interest. An 
+    AR(p) model with order p = 1 may be written as X(t) = A(1) * X(t-1) + E(t),
+    where
+        :X(t) is the time series under investigation 
+        :A(1) is the autoregressive parameter
+        :X(t-1) is the time series lagged 1 period
+        :E(t) is the error term of the model or white noise
+
+    In other words, any value in X(t) can be explained using a linear 
+    combination of the past value T(t-1) plus some error term E(t). X(t)
+    could also be a linear combination of more than one past value: 
+    X(t) = A(1) * X(t-1) + A(2) * X(t-2) + E(t).
+
+    Differencing is a way of making a non-stationary time series stationary. This 
+    is done by computing the differences between consuective observations 
+    (subtracting the observation from the current period from the previous one).
+    Differencing can help stabilize the mean of a time series by removing changes 
+    in the level of a time series, which reduces trend and seasonality. If the 
+    transformation is done once, then the data has been "first differenced". The 
+    same transformation can be done again, so the data would be "second 
+    differenced".
+
+    Moving average components uses past forecast errors E(t). In other words,
+    X(t) can be thought of as a weighted moving average of the past forecast 
+    errors: X(t) = c + E(t) + W(1)*E(t-1) + ... + W(q)*E(t-q).
+    """
+
+    def __init__(self):
+        self.best_params = {}
+    
+
+    def _evaluate_arima_model(self, X: Union[pd.Series, pd.DataFrame], arima_order: Tuple[int, int, int], train_size: float, freq: str) -> Tuple[float, dict]:
```

Comment:
train size should be an int

Reply:
I changed it so it could take in a float, int or None (similar to what scikit-learn does). 

- If float, should be between 0.0 and 1.0 and represent the proportion of the dataset to include in the train split. If int, represents the absolute number of train samples. If None, the value is automatically set 0.75 (https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html).

Is that too complicated or should we just use int to simplify things